### DISCUSSION TITLE
Cannot get session in Private Browsing Mode

### DISCUSSION THREAD

**Discussion by phamhungvn** [nextauthjs/next-auth#4544] [Help]
### Question 💬

Hi,
I'm developing a Next-Auth App with IdentityServer4. 
Everything seems to be working fine until I switch to **Private Browsing Mode (Incognito Chrome windows).**
In **Private Browsing Mode**, I couldn't get session anymore.

I'm facing the same issue  with **Safari on Mac**.
 So, I wonder if it's the known-issue of Next-Auth or I have configured something wrong.

I have checked the Cookies and saw that, in **Private Browsing Mode**, some values are not available 

I have attached here the screenshots:

![NextAuth Cookies](https://user-images.githubusercontent.com/85485496/167407938-2fc1bdb9-2b99-4f4f-9a7a-2864fba29792.png)

![next-auth session token](https://user-images.githubusercontent.com/85485496/167411215-95a95975-6695-4038-a717-b8ab62b8e9b8.png)

As you can see in the Firefox console log: Cookie "next-auth.session-token" has been rejected because it is already expired. I think this is the reason why the cookie "next-auth.session-token" is not available and so I cannot get session.

I have attached here sample code. Pls check it and give me advice!
Many thanks!
[NextAuth Sample.zip](https://github.com/nextauthjs/next-auth/files/8651718/NextAuth.Sample.zip)

### How to reproduce ☕️

Run next-auth sample in Private Mode

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by phamhungvn** [comment]
In Safari, if I disable **"Prevent cross-site tracking"**, the cookie "**next-auth.session-token**" is available and it will run well. So, I think the root cause is: Next-Auth couldn't store cookie "**next-auth.session-token**" in Private Mode or if enable **"Prevent cross-site tracking"** in Safari.

Need your advice on how to overcome this issue.
Many thanks!

**Comment by ThangHuuVu** [comment]
Unfortunately, I couldn't reproduce this on our example app: https://next-auth-example.vercel.app/. I still can log in and cookies are set.
So I suspect a configuration error. Perhaps it's related to the custom provider you are using?
Could you set up a minimal reproduction to demonstrate it and share it here? I've read the code you attached but didn't see anything suspicious.

**Reply by phamhungvn** [reply]
Thank you very much for your feedback @ThangHuuVu !
I already re-checked and **YES**, cookies are set but the issue is still there.
I'm using Next-Auth and IdentityServer4 (ID4) to implement SSO (Single-Sign-On) for my project and because Next-Auth doesn't support Federated Sign-out so I have to add some customizations:
- Perform monitoring on the check_session_iframe to notify clients that the user has signed-out. This is where the issue arises:
The RP iframe polls the OP iframe with postMessage Client ID + " " + Session State and has to be able to receive the postMessage back from the OP iframe. The received data will either be "**changed**" or "**unchanged**" or "**error**". If the received data is "**changed**', the user will be signed-out.
In Normal Mode, everything goes well but in Private Mode, the received data is always "**changed**" and it causes trouble: The user will be signed-out few seconds after he signed-in successfully. It's because the message from OP iframe is always "changed' then the user will be signed-out.
Can you please advise me on this? I can send you the project source code if needed.
Many thanks!

**Comment by zakwear** [comment]
Also getting this behavior when using an iframe from Stripe. Console warns:

`Partitioned cookie or storage access was provided to “https://m.stripe.network/inner.html#url=http%3A%2F%2Flocalhost%3A3000%2Fonboarding%2Fcreate&title=&referrer=http%3A%2F%2Flocalhost%3A3000%2Fauth%2Fsignin%3FcallbackUrl%3Dhttp%253A%252F%252Flocalhost%253A3000%252Fonboarding%252Fcreate%26error%3DSessionRequired&<more idData>” because it is loaded in the third-party context and dynamic state partitioning is enabled.`

After some time the cookie expires, and HTTP requests to db/apis fail.

If it continues I'll make a codesandbox.

**Comment by basememara** [comment]
Did you ever get this resolved? I'm seeing a similar issue with Safari Private Browsing too but with magic email link. The problem is that in Safari Private mode, each tab is its own session. When a user initiates the login and checks their email, the link opens a new tab but doesn't have the cookies they started out with so the `callbackUrl` isn't shared. The login completes but other things are broken.

You can see the contrast with Google Chrome Incognito. If you log into some website in one tab, opening a new tab shares that session. Safari Private Browsing has a separate session per tab.
